src/fwds_rvs_data.jl

```diff
@@ -860,15 +860,14 @@ tangent type. This method must be equivalent to `tangent_type(_typeof(primal))`.
 @foldable tangent_type(::Type{NoFData}, ::Type{R}) where {R<:IEEEFloat} = R
 @foldable tangent_type(::Type{F}, ::Type{NoRData}) where {F<:Array} = F
 
-# Union types. NOTE: Only `Union{Nothing, Array{<:Base.IEEEFloat}, Base.IEEEFloat}` are supported. 
 @foldable function tangent_type(
     ::Type{NoFData}, ::Type{R}
-) where {R<:Union{NoRData,T} where {T<:Base.IEEEFloat}}
+) where {R<:Union{NoRData,T} where {T<:Any}}
```

Comment:
These types are too loose and might bite us later. 

 Here `T` should be subject to constraint `rdata_type(T) = NoRData`, but it is hard to come up with a type constraint for this requirement beyond `IEEEFloat`